src/libstd/net/ip.rs

```diff
@@ -122,8 +167,23 @@ impl IpAddr {
     }
 
     /// Returns true if this address is in a range designated for documentation ([IPv4], [IPv6]).
+    ///
     /// [IPv4]: ../../std/net/struct.Ipv4Addr.html#method.is_documentation
     /// [IPv6]: ../../std/net/struct.Ipv6Addr.html#method.is_documentation
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(ip)]
+    ///
+    /// use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+    ///
+    /// fn main() {
```

Comment:
You should be able to omit the `fn main() {` even if you're using `#![feature(...)]`. An example:

https://github.com/rust-lang/rust/blob/b30022a1d32ebbd41f6623ab6fe7c62d489548e1/src/libcore/sync/atomic.rs#L205-L213